spec/enumerable_spec.rb

```diff
@@ -0,0 +1,160 @@
+require('./lib/enumerable.rb')
+include Enumerable
+
+class MockObject
+  attr_reader :variable, :other_variable
+  def initialize variable = 'hello', other_variable = 'world'
+    @variable = variable
+    @other_variable = other_variable
+  end
+end
+
+RSpec.describe Enumerable do
+  describe "#my_each" do
+    it "should access all element in given array" do
+      counter = 0
+      array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```

Comment:
[Use let or let! when defining test subjects ](http://www.betterspecs.org/#let)

Reply:
We used it but as I mentioned in tic-tac-toe, I think that this idea doesn't fit at all for this project. If we use some service layers as a object something like FooCalculatorService, we can use this declarations but let me say that we need to specific arrays for some scenarios for example positive_numbers or negative_numbers